client/src/store/usePlayer.ts

```diff
@@ -23,3 +23,12 @@ export function useMe(): string {
   }
   return myName;
 }
+
+export function useHasJoinedGame(): [boolean, (key: boolean) => void] {
```

Comment:
I think we should make this universal, just always prepopulate with the last name you used to join any game from this device.

Reply:
Oh, this is different.  this is a hack, we should not do this.

Reply:
Instead, we should enter a loading state until the player is in the game.

Reply:
Actually I just use it to know if a player is kicked or not :/

Reply:
Otherwise it's difficult to determine the difference if we want to keep the player message on join without adding another field to our game instance.

Reply:
the player has been kicked IF:
- they have a key for the game
- the game does not contain their name
AND
the join game request has finished.

We should just put ourselves in a loading state while that request is going.

Reply:
But also whateve3r let's leave this hack and just revamp joining later after we fix game status